lib/watir/wait.rb

```diff
@@ -246,5 +246,24 @@ def wait_while_present(timeout = nil)
       # it's not present
     end
 
+    #
+    # Waits until the element is present.
+    #
+    # @example
+    #   browser.text_field(name: "new_user_first_name").wait_until_text
+    #
+    # @param [Fixnum] timeout seconds to wait before timing out
+    # @param [String] text value to wait for before timing out
+    #
+    # @see Watir::Wait
+    # @see Watir::Element#text
+    #
+
+    def wait_until_text(text, timeout = nil)
+      timeout ||= Watir.default_timeout
+      message = "waiting for #{selector_string} text to equal \"#{text}\""
+      Watir::Wait.until(timeout, message) { self.text =~ /#{text}/ }
```

Comment:
We want to be able to support both Regexp match and String match. This seemed like an easy way to do that.